src/layouts/EditContactUs.jsx

```diff
@@ -185,8 +196,13 @@ export default class EditContactUs extends Component {
         },
       });
     } catch (err) {
+      if (!this.state.shouldRedirectToNotFound) {
```

Comment:
We shouldn't rely on a state check here since state updating is asynchronous and it's possible that by the time the code reaches this block, the state has not yet been updated.

Reply:
Good point, resolved in https://github.com/isomerpages/isomercms-frontend/pull/290/commits/ba0aab2a90d109ca05267c110db67a6efd3b9baf